cli.py

```diff
@@ -0,0 +1,166 @@
+
+        else:                                                          #if file name is provided read string from file
+            while not bech32m.read_file(infile, input_format):
+                infile = input("Enter input file name:")
+
+            input_string = bech32m.read_file(infile, input_format)
+
+        #validate input string with the input format provided, if string not in format ask for input string again
+        if input_format == "b64":
+            while not bech32m.check_if_str_is_base64(input_string):
+                input_string = input("Enter Valid \"base64\" input string:")
+
+        if input_format == "hex":
+            while not bech32m.check_if_str_is_hex(input_string):
+                input_string = input("Enter valid \"Hex\" input string:")
+
+        if input_format == "bin":
+            while not bech32m.check_if_str_is_bin(input_string):
+                input_string_s = input("Enter valid \"bin\" input string:")
+                #input_string = bytes(input_string_s, 'utf-8')
+                input_string = input_string_s.encode()
+
+        #input output format and validate
+        output_format = input("Choose output Format[b64/bin/hex]"
+                              " (Press Enter for Default[\"bech32m\" for \"enc\" and \"hex\" for \"dec\"]):")
+        while output_format not in ["b64", "bin", "hex", ""]:
+            output_format = input("Enter Valid Output Format[b64/bin/hex]"
+                                  "(Press Enter for Default[\"hex\" for \"enc\" and \"bech32m\" for \"dec\"]):")
+
+        #set default output format for enc/dec if not provided
+        if mode == "enc" and output_format == "":
+            output_format = "bech32m"
+        if mode == "dec" and output_format == "":
+            output_format = "hex"
+
+        #input output file name if enc/dec data is to be written to a file
+        outfile = input("Enter output file name (press enter to print output to \"stdout\") : ")
+
+        return mode, hrp, input_string, infile, input_format, output_format, outfile
+
+
+def main():
+    mode, hrp, input_string, input_file, input_format, output_format, output_file = user_interface()
+
+    if input_string is None:
+        if not bech32m.read_file(input_file, input_format):
+            sys.exit(0)
+        else:
+            input_string = bech32m.read_file(input_file, input_format)
+
+    if mode == "enc":
+        if input_format is None:
+            input_format = "hex"
+        if output_format is None:
+            output_format = "bech32m"
+
+        print("Mode:", mode, "HRP:", hrp, "Input String:", input_string, "Input Format:", input_format,
+              "Output Format:", output_format)
+
+        encoded_string = enc_dec.encode_pure_bech32m(hrp, input_string, input_format, output_format)
+
+        if not output_file:
+            print("\nEncoded String in {}".format(output_format[:].upper()), encoded_string)
```

Comment:
Output like this makes it very hard to use this application by a different program